File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> numMap = new HashMap<>();
        int[] res= new int[2];
        int diff = 0;
        for (int i = 0; i < numbers.length; i++) {
            diff = target - numbers[i];
            if(numMap.containsKey(diff)){
                res[0]= numMap.get(diff)+1;
                res[1] = i+1;
                return res;
            }
            numMap.put(numbers[i],i);

        }
        return res;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """

        # USE POINTERS
        l,r= 0, len(numbers)-1
        res=[]

        while l<r:
            summ = numbers[l]+numbers[r]
            if summ == target:
                return [l+1,r+1]
            elif summ<target:
                l+=1
            else:
                r-=1
        
        



        # res =[]
        # HashSet hs = set<int>
        # for n in numbers:
        #     diff = target- n
        #     if diff in numbers:
        #         res = [numbers.index(diff)+1,numbers.index(n)+1]
        #     if diff in numbers and diff == n:
        #         res = [numbers.index(n),numbers.index(n)]
        # return res


        